cameo/core/metabolite.py

```diff
@@ -53,6 +54,45 @@ def constraint(self):
         else:
             return None
 
+    def _relax_mass_balance_constrain(self, time_machine, k):
+        if time_machine:
+            time_machine(do=partial(setattr, self.constraint, "lb", -k),
+                         undo=partial(setattr, self.constraint, "lb", self.constraint.lb))
+            time_machine(do=partial(setattr, self.constraint, "ub", k),
+                         undo=partial(setattr, self.constraint, "ub", self.constraint.ub))
+        else:
+            self.constraint.lb = -k
+            self.constraint.ub = k
+
+    def knock_out(self, time_machine=None, k=10000):
+        """'Knockout' a metabolite [1].
+
+        Implementation according to the paper:
+            "All fluxes around the metabolite M should be restricted to only produce the metabolite,
+             for which balancing constraint of mass conservation is relaxed to allow nonzero values
+             of the incoming fluxes whereas all outgoing fluxes are limited to zero."
+
+        Parameters
+        ----------
+        time_machine : TimeMachine
+            An action stack to reverse actions.
+        k: number
+            A large number
+
+        References
+        ----------
+        .. [1] Kim, P.-J., Lee, D.-Y., Kim, T. Y., Lee, K. H., Jeong, H., Lee, S. Y., & Park, S. (2007).
+          Metabolite essentiality elucidates robustness of Escherichia coli metabolism. PNAS, 104(34), 13638-13642
+        """
+        # restrict reactions to produce metabolite
+        for reaction in self.reactions:
+            if reaction.metabolites[self] > 0:  # for positive stoichiometric coefficient set lb to 0
+                reaction.change_bounds(lb=0, time_machine=time_machine)
+            elif reaction.metabolites[self] < 0:  # for negative stoichiometric coefficient set ub to 0
+                reaction.change_bounds(ub=0, time_machine=time_machine)
```

Comment:
I would still argue that the new constraint (no flux through reactions that consume the metabolite) should be added on top of existing constraints (at least 50 flux through reaction) instead of overriding it silently. The same should be the case for knocking out a reaction imo, even though I know that's not how it currently behaves.